presentation/backend/src/presentation-backend/PresentationManagerDetail.ts

```diff
@@ -0,0 +1,396 @@
+import { getElementKey, getLocalesDirectory } from "./Utils";
+
+/** @internal */
+export class PresentationManagerDetail implements IDisposable {
+  private _disposed: boolean;
+  private _nativePlatform: NativePlatformDefinition | undefined;
+  private _updatesTracker: UpdatesTracker | undefined;
+  private _onManagerUsed: (() => void) | undefined;
+
+  public rulesets: RulesetManager;
+  public activeLocale: string | undefined;
+  public activeUnitSystem: UnitSystemKey | undefined;
+
+  constructor(params: PresentationManagerProps) {
+    this._disposed = false;
+
+    const presentationAssetsRoot = params.presentationAssetsRoot ?? {
+      common: PRESENTATION_COMMON_ASSETS_ROOT,
+      backend: PRESENTATION_BACKEND_ASSETS_ROOT,
+    };
+    const mode = params.mode ?? PresentationManagerMode.ReadWrite;
+    const changeTrackingEnabled = mode === PresentationManagerMode.ReadWrite && !!params.updatesPollInterval;
+    this._nativePlatform = params.addon ?? createNativePlatform(
+      params.id ?? "",
+      typeof presentationAssetsRoot === "string" ? presentationAssetsRoot : presentationAssetsRoot.common,
+      params.localeDirectories ?? [],
+      params.workerThreadsCount ?? 2,
+      mode,
+      changeTrackingEnabled,
+      params.caching,
+      params.defaultFormats,
+      params.useMmap,
+    );
+
+    const getNativePlatform = () => this.getNativePlatform();
+    if (IpcHost.isValid && changeTrackingEnabled) {
+      this._updatesTracker = UpdatesTracker.create({
+        nativePlatformGetter: getNativePlatform,
+        pollInterval: params.updatesPollInterval!,
+      });
+    } else {
+      this._updatesTracker = undefined;
+    }
+
+    setupRulesetDirectories(
+      this._nativePlatform,
+      typeof presentationAssetsRoot === "string" ? presentationAssetsRoot : presentationAssetsRoot.backend,
+      params.supplementalRulesetDirectories ?? [],
+      params.rulesetDirectories ?? [],
+    );
+    this.activeLocale = params.defaultLocale;
+    this.activeUnitSystem = params.defaultUnitSystem;
+
+    this._onManagerUsed = undefined;
+    this.rulesets = new RulesetManagerImpl(getNativePlatform);
+  }
+
+  public dispose(): void {
+    if (this._disposed) {
+      return;
+    }
+
+    this.getNativePlatform().dispose();
+    this._nativePlatform = undefined;
+
+    this._updatesTracker?.dispose();
+    this._updatesTracker = undefined;
+
+    this._disposed = true;
+  }
+
+  public getNativePlatform(): NativePlatformDefinition {
+    if (this._disposed) {
+      throw new PresentationError(
+        PresentationStatus.NotInitialized,
+        "Attempting to use Presentation manager after disposal",
+      );
+    }
+
+    return this._nativePlatform!;
+  }
+
+  public setOnManagerUsedHandler(handler: () => void) {
+    this._onManagerUsed = handler;
+  }
+
+  public async getContentDescriptor(requestOptions: Prioritized<ContentDescriptorRequestOptions<IModelDb, KeySet>>): Promise<string> {
```

Comment:
Why `getContentDescriptor` and `compareHierarchies` have their own methods here, but in all other cases `request` is called directly?

Reply:
No good reason, moved it back in 6aaa665bdd80a711afa8f3636cd08d8594fc7417.